File: Strings/pagina03/2017/2017.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").split("\n")

const hamming_distance = (str1, str2) => {
    let distancie = 0
    for(let i = 0; i < str1.length; i++){
        if(str1[i] != str2[i]) distancie++
    }
    return distancie
}

const str1 = input.shift()
const k = Number(input.shift())

let indice = -1
let menor_distancia = Infinity

for(let i = 0; i < 5; i++){
    const str2 = input.shift()
    const distance = hamming_distance(str1, str2)

    if(distance < menor_distancia){
        menor_distancia = distance
        indice = i + 1
    }
}

console.log(indice)
console.log(menor_distancia <= k ? menor_distancia : -1)

File: Strings/__2000__/2017/2017.py

def hamming_distance(s1, s2):
    distancia = 0
    for i in range(len(s1)):
        if s1[i] != s2[i]:
            distancia += 1
    return distancia
    

string_principal = input()
k = int(input())

indice = -1
menor_distancia = float('inf')

for i in range(5):
    palavra = input()
    distancia = hamming_distance(string_principal, palavra)
    
    if distancia < menor_distancia:
        menor_distancia = distancia
        indice = i + 1

print(indice)

print(menor_distancia if menor_distancia <= k else -1)